code/backtracking/m_coloring_problem/m_coloring.cpp

```diff
@@ -0,0 +1,59 @@
+#include <std/c++.h>
+using namespace std;
+
+bool graph[500][500];	//globally defined adjacency list
```

Comment:
avoid define in global.
I suggest that you could use vector or dynamic array pointer, and let is as parameter which will be flexible.